viewer/packages/camera-manager/src/CameraManager.ts

```diff
@@ -1,604 +1,68 @@
-    }
-    if (this._onWheel !== undefined && removeOnWheel) {
-      this._domElement.removeEventListener('wheel', this._onWheel);
-      this._onWheel = undefined;
-    }
-  }
-
-  private handleMouseWheelActionChange(controlsOptions: CameraControlsOptions) {
-    const { _controls: controls } = this;
-
-    switch (controlsOptions?.mouseWheelAction) {
-      case 'zoomToTarget':
-        controls.zoomToCursor = false;
-        break;
-      case 'zoomPastCursor':
-        controls.useScrollTarget = false;
-        controls.zoomToCursor = true;
-        break;
-      case 'zoomToCursor':
-        controls.setScrollTarget(controls.getState().target);
-        controls.useScrollTarget = true;
-        controls.zoomToCursor = true;
-        break;
-      case undefined:
-        break;
-      default:
-        assertNever(controlsOptions.mouseWheelAction);
-    }
-  }
+  on(event: 'cameraChange', callback: CameraChangeData): void;
   /**
-   * Method for setting up camera controls listeners and values inside current controls class.
+   * Unsubscribes from changes of the camera event.
+   * @param event Name of the event.
+   * @param callback Callback function to be unsubscribed.
    */
-  private setupControls() {
-    let scrollStarted = false;
-
-    const wheelClock = new THREE.Clock();
-
-    const onClick = async (e: any) => {
-      this._controls.enableKeyboardNavigation = false;
-      const newTarget = await this.calculateNewTarget(e);
-      this.setCameraTarget(newTarget, true);
-    };
-
-    const onWheel = async (e: any) => {
-      const timeDelta = wheelClock.getDelta();
-
-      if (timeDelta > CameraManager.DefaultMinimalTimeBetweenRaycasts) scrollStarted = false;
-
-      const wantNewScrollTarget = !scrollStarted && e.deltaY < 0;
-      const isZoomToCursor = this._cameraControlsOptions.mouseWheelAction === 'zoomToCursor';
-
-      if (wantNewScrollTarget && isZoomToCursor) {
-        scrollStarted = true;
-        const newTarget = await this.calculateNewTarget(e);
-        this._controls.setScrollTarget(newTarget);
-      }
-    };
-
-    if (this._controls) this.handleMouseWheelActionChange(this._cameraControlsOptions);
-
-    if (this._cameraControlsOptions.changeCameraTargetOnClick && this._onClick === undefined) {
-      this._inputHandler.on('click', onClick);
-      this._onClick = onClick;
-    }
-    if (this._onWheel === undefined) {
-      this._domElement.addEventListener('wheel', onWheel);
-      this._onWheel = onWheel;
-    }
-  }
+  off(event: 'cameraChange', callback: CameraChangeData): void;
 
+  fitCameraToBoundingBox(boundingBox: THREE.Box3, duration?: number, radiusFactor?: number): void;
   /**
-   * Convert pixel coordinates of the cursor to [-1,1]^2 coordinates.
-   * @param pixelX
-   * @param pixelY
+   * Updates internal state of camera manager. Expected to update visual state of the camera
+   * as well as it's near and far planes if needed. Called every time screen is refreshed.
```

Comment:
`Called every time screen is refreshed.` <- this is wrong, right? It's actually called more often than that?

Reply:
That's correct and one of the things that should be changed, I believe. Also, it's not every new rendered frame but every screen refresh with `requestAnimationFrame`.

Reply:
The doc should reflect what actually happens. I at least read this "every time the 3D view is re-rendered". A more precise version might be "Called in requestAnimationFrame-loop" and also emphasis that it's very important that the performance is high.

Reply:
Changed description to: 
` /**
   * Updates internal state of camera manager. Expected to update visual state of the camera
   * as well as it's near and far planes if needed. Called in `requestAnimationFrame`-loop. 
   * Reveal performance affects frequency with which this method is called.
   * @param deltaTime Delta time since last update in seconds.
   * @param boundingBox Global bounding box of the model(s) and any custom objects added to the scene.
   */
  `